### DISCUSSION TITLE
[RFC][Button] Параметр для задания отрицательных отступов

### DISCUSSION THREAD

**Discussion by inomdzhon** [VKCOM/VKUI#2491] [Ideas] [answered]
## Задача

В [Banner](https://vkcom.github.io/VKUI/#/Banner) кнопка [Button mode="tertiary"](https://vkcom.github.io/VKUI/#/Button) должен быть смещён влево, чтобы выравнить её по левому краю.

## Как сейчас?

Сейчас это сделано через каскад

https://github.com/VKCOM/VKUI/blob/289eda6048ec1b439b21d2bc3bb5fde42237b85d/src/components/Button/Button.css#L545-L548

![1c68-1936-d46e-2150](https://user-images.githubusercontent.com/5850354/168804638-81fa8c1c-8237-47cb-b9a5-c777ca831aaa.png)

## Как надо?

По возможности мы уходим от обращения из одного блока к другому, поэтому в рамках задачи https://github.com/VKCOM/VKUI/issues/2469 я удаляю каскад от [Banner](https://vkcom.github.io/VKUI/#/Banner).

Соотвественно столкнулся с проблемой, что теперь [Button mode="tertiary"](https://vkcom.github.io/VKUI/#/Button) не выравнивается по левому краю

![image](https://user-images.githubusercontent.com/5850354/168805871-d6e76f1d-f2a7-4481-84d5-b2a7c59d3973.png)

### Возможные решения

> Во всех вариантах пользователь сам будет регулировать выравнивание по левому краю внутри `Banner`.

1. Добавить [Button](https://vkcom.github.io/VKUI/#/Button) параметр `negXOffset`, который будет задавать отрицательный `margin` равный `padding` слева и справа.
2. Добавить [Button](https://vkcom.github.io/VKUI/#/Button) параметр `noXGap`, который будет обнулять `padding` слева и справа. ⚠️ при `hasHover={true}` будет некрасивое при наведении
3. Создать новый компонент :) Типа как `Button`, но без отступов слева и справа и со своим поведением на наведения

---

Я пока склоняюсь к первому. Что думаете? Видите ли иные способы решения?

**Comment by Fliqle** [comment]
Из предложенных я тоже за 1. Возможно еще пригодится где-нибудь

**Comment by inomdzhon** [comment]
## Summary

Решили отказаться [Button mode="tertiary"](https://vkcom.github.io/VKUI/#/Button) внутри [Banner](https://vkcom.github.io/VKUI/#/Banner).

Вместо него, следует использовать [Button mode="secondary"](https://vkcom.github.io/VKUI/#/Button)

**Comment by inomdzhon** [comment, marked as answer]
Будет создан новый тип кнопки, у которого

1. Нет боковых отступов
2. Hover/Active меняют цвет шрифта

**Reply by inomdzhon** [reply]
Добавлен `mode="link"` (см. https://vkcom.github.io/VKUI/5.9.3/#/Button)
